docs/DeveloperGuide.md

```diff
@@ -4,9 +4,118 @@
+
+#### The `WorkoutLog` class
+A `WorkoutLog` object is similar to that of a `Workout` object, except it contains an `ArrayList` of `ExerciseLog` objects.
+
+### <code>ActivityManager</code> and child classes
+The <code>ActivityManager</code> and inheritors are responsible for managing an <code>ArrayList</code> of activities. The basic functions of an <code>ActivityManager</code> include:
+1. <code>add()</code>: Adding an activity to the <code>ArrayList</code>
+2. <code>delete()</code>: Deleting an activity from the <code>ArrayList</code>
+3. <code>retrieve()</code>: Retrieving an activity from the <code>ArrayList</code> by name
+4. <code>getListString()</code>: Get the string containing all the activities contained in the <code>ActivityManager</code>.
+5. `execute()`: Execute all commands related to the `ActivityManager` and return the required user input.
+
+![ActivityManagerClassDiagram](diagrams/ActivityManagerClassDiagram.png)
+
+#### The `ExerciseManager` class
+`ExerciseManager` is responsible for tracking and manipulating all exercises added to `ByteCeps` by the user.
+
+#### The `WorkoutManager` class
+`WorkoutManager` is responsible for tracking and manipulating all workouts created by the user.
+
+#### The `WeeklyProgramManager` class
+`WeeklyProgram` is responsible for tracking and manipulating the weekly training program set by the user.
+
+## Implementation
+#### Add an Exercise
+1. The process begins with the user inputting a command via the command-line interface. In this scenario, the user enters the command `exercise /add pushups`, indicating their intention to add a new exercise named `pushups` to the system.
+2. The command parser receives the user input and parses it to extract relevant information, such as the action (`exercise /add`) and any additional parameters (e.g., the name of the exercise to be added).
+3. Upon receiving the parsed command, the Exercise Manager component validates the input to ensure it conforms to the expected format and criteria. This step is crucial for maintaining data integrity and preventing errors in subsequent processing.
+4. If the input passes validation, the Exercise Manager proceeds to create a new Exercise object with the specified name, `pushups`. It then invokes the add method within the Activity Manager component to add the newly created Exercise to the activity set. Finally, a success message confirming the addition of the exercise is printed to the user interface, indicating that the operation was completed successfully.
+5. If the input fails validation, an error message is generated and displayed to the user, informing them of the invalid command format. This ensures that users receive timely feedback and can correct their input accordingly
+
+The sequence diagram below shows how an exercise is created.
+
+![addExercise](https://github.com/V4Vern/tp/assets/28131050/45f7e9b3-8a31-4dfe-a783-433acb71fa58)
+
```

Comment:
Should there be an extra box coming out since you are calling a method of its own class?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110594538/ba5ed18b-e952-4ba4-917d-7ed90ee7f6fb)